outputs.tf

```diff
@@ -63,6 +63,11 @@ output "kubeconfig_filename" {
   value       = concat(local_file.kubeconfig.*.filename, [""])[0]
 }
 
+output "oidc_provider_arn" {
+  description = "The ARN of the OIDC Provider if `enable_irsa = true`."
+  value       = var.enable_irsa ? aws_iam_openid_connect_provider.oidc_provider : null
```

Comment:
This isn't returning just the ARN. It's returning a list of `aws_iam_openid_connect_provider` outputs.

For example:
```
oidc = [
  {
    "arn" = "arn:aws:iam::111111111111:oidc-provider/oidc.eks.us-west-2.amazonaws.com/id/518F3C3FCEEB9B22064F6EE41A9BDC39"
    "client_id_list" = [
      "sts.amazonaws.com",
    ]
    "id" = "arn:aws:iam::111111111111:oidc-provider/oidc.eks.us-west-2.amazonaws.com/id/518F3C3FCEEB9B22064F6EE41A9BDC39"
    "thumbprint_list" = [
      "9e99a48a9960b14926bb7f3b02e22da2b0ab7280",
    ]
    "url" = "oidc.eks.us-west-2.amazonaws.com/id/518F3C3FCEEB9B22064F6EE41A9BDC39"
  },
]
```

```suggestion
  value       = var.enable_irsa ? aws_iam_openid_connect_provider.oidc_provider[0].arn : null
```

Tried with 0.12.9 and 0.12.18. Appears to work correctly. Having indexes in outputs scare me a little due to historical issues but it seems the lazy evaluation of the ternary operator is working now. Also tried manually deleting the resource, and running destroy on an empty state with `enable_irsa = true`.

Setting an output to `null` causes it to not be listed in the state's output. This may cause issues for people using the `terraform_remote_state` data source to chain configs together.

Reply:
Good catch and suggested change, I'll push an update. I'm also not a fan of how using the ternary operator plus count forces you to refer to an index and have been pondering whether switching to a for_each makes more sense, however the place to make that switch is where the resource is being created. Do you think that is the smarter way to do it?

Setting up `terraform_remote_state` will work regardless of whether `enable_irsa` is true or false, however anything referencing this output _will_ fail when it is set to false. I believe that **this is desirable behavior** since an early Terraform error message is preferable to a resource being created with bad data downstream. Say for example that we use the empty string when `enable_irsa = false` and the user attempts to deploy a Kubernetes manifest with this value. The user would likely see no errors in Terraform and end up wasting a bunch of time trying to debug it as a Kubernetes problem.